### PR TITLE
[merlin] Add option to `dune ocaml-merlin` to dump generic config in dot merlin syntax

### PR DESCRIPTION
**Pull request by voodoos**
This is part of #4217 

This PR adds an option to `dune ocaml-merlin` to print a generic merlin file on stdout.

It does a rough merging of the configurations found in the folder. 
Preprocessing configuration and flags are commented out. It is up to the user to choose what to keep.

This is not meant to generate the best approximation as before. Rather it is an escape hatch for users with non-standard configurations.

The path to the directory is optional:
`dune ocaml-merlin --dump-dot-merlin`,
`dune ocaml-merlin --dump-dot-merlin="."`,
`dune ocaml-merlin --dump-dot-merlin="$(pwd)"` 
will all print the configuration of the current directory.

### CODE REVIEW

File: bin/ocaml_merlin.ml [hunk 1]
```diff
@@ -20,13 +20,21 @@ let term =
...
              debugging purposes only and should not be considered as a stable \
              ouptut.")
+  and+ dump_dot_merlin =
+    Arg.(
+      value
+      & opt ~vopt:(Some ".") (some string) None
+      & info [ "dump-dot-merlin" ]
+          ~doc:
+            "Prints a coarse merge of all merlin configuration for the given \
+             folder in the Merlin configuration syntax.")
```
Comment (rgrinberg) [inline bin/ocaml_merlin.ml:37]: Can you change this command to be `dune ocaml dump-dot-merlin` instead?

File: bin/ocaml_merlin.ml [hunk 2]
```diff
@@ -43,3 +43,47 @@ let term =
...
+      Arg.(
+        value
+        & pos 0 (some string) None
+        & info [] ~docv:"PATH"
+            ~doc:
+              "The path to the folder of which the configuration should be \
+               printed. Defaults to the current directory.")
+    in
+    Common.set_common common ~log_file:No_log_file ~targets:[];
+    Scheduler.go ~common (fun () ->
```
Comment (rgrinberg) [inline bin/ocaml_merlin.ml:75]: if you aren't using fibers, perhaps you don't need the scheduler at all?

**Review by rgrinberg — APPROVED** [review]

**Review by rgrinberg — APPROVED** [review]

**Comment by rgrinberg** [comment]
I'm not sure if this can go into 2.8.3. This looks more like a new feature to me.

**Comment by voodoos** [comment]
CI failure is unrelated to this PR, I will rebase and merge.

**Pull request merged**